mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -292,6 +296,89 @@ emulatedVectorLoad(OpBuilder &rewriter, Location loc, Value base,
       newLoad);
 }
 
+/// Selects values from two sources based on a mask, and casts the result to a
+/// new type.
+static Value selectAndCast(OpBuilder &builder, Location loc,
+                           VectorType castIntoType, Value mask, Value trueValue,
+                           Value falseValue) {
+  Value maskedValue =
+      builder.create<arith::SelectOp>(loc, mask, trueValue, falseValue);
+  return builder.create<vector::BitCastOp>(loc, castIntoType, maskedValue);
+}
+
+/// Emits `memref.generic_atomic_rmw` op to store a subbyte-sized value to a
+/// byte in memory, with a mask. The `valueToStore` is a vector of subbyte-sized
+/// elements, with size of 8 bits, and the mask is used to select which elements
+/// to store.
+///
+/// Inputs:
+///   linearizedMemref = |2|2|2|2| : <4xi2> (<1xi8>)
+///   linearizedIndex = 2
```

Comment:
Is my understanding correct?

* `linearizedMemref` and `valueToStore` are 1-D vectors (if this is true, could you add an assert for the memref?)
* `linearizedIndex` is the index into the 1-D input memref? If yes, I suggest `linarizedIndex` -> `storeIdx` (this method doesn't really "know" what "linearization" means).

Reply:
yes, you are correct! :-)

updated `storeIdx` as suggested.

Reply:
> updated storeIdx as suggested.

Looks like you forgot to push?

Reply:
@banach-space duh, I did forget about it. pushed now.